packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/model/RemoteNotificationContent.kt

```diff
@@ -0,0 +1,99 @@
+package expo.modules.notifications.notifications.model
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.os.Parcel
+import android.os.Parcelable
+import com.google.firebase.messaging.RemoteMessage
+import expo.modules.notifications.notifications.enums.NotificationPriority
+import expo.modules.notifications.notifications.interfaces.INotificationContent
+import expo.modules.notifications.notifications.presentation.builders.downloadImage
+import org.json.JSONObject
+
+/**
+ * A POJO representing a remote notification content: title, message, body, etc.
+ * The content originates in a RemoteMessage object.
+ *
+ * Instances of this class are not persisted in SharedPreferences (unlike {@link NotificationContent}. This class does
+ * not implement Serializable, but Parcelable ensures we can pass instances between different parts of the application.
+ */
+class RemoteNotificationContent(private val remoteMessage: RemoteMessage) : INotificationContent {
+
+  constructor(parcel: Parcel) : this(parcel.readParcelable<RemoteMessage>(RemoteMessage::class.java.classLoader)!!)
+
+  override suspend fun getImage(context: Context): Bitmap? {
+    val uri = remoteMessage.notification?.imageUrl
+    return uri?.let { downloadImage(it) }
+  }
+
+  override fun containsImage(): Boolean {
+    return remoteMessage.notification?.imageUrl != null
+  }
+
+  override val title: String?
+    get() = remoteMessage.notification?.title
+
+  override val text: String?
+    get() = remoteMessage.notification?.body
+
+  override val shouldPlayDefaultSound: Boolean
+    get() = remoteMessage.notification?.sound == null
+
+  override val soundName: String?
+    get() = remoteMessage.notification?.sound
+
+  override val shouldUseDefaultVibrationPattern: Boolean
+    get() = remoteMessage.notification?.defaultVibrateSettings == true
+
+  override val vibrationPattern: LongArray?
+    get() = remoteMessage.notification?.vibrateTimings
+
+  override val body: JSONObject?
+    get() = try {
+      JSONObject(remoteMessage.data["body"] ?: "{}")
```

Comment:
Are you sure that we want to return an empty object instead of `null` when `body` is missing?

Reply:
thanks, fixed that